python/ray/data/_internal/plan.py

```diff
@@ -647,6 +647,17 @@ def execute(
                     reply.store_stats.restored_bytes_total
                 )
 
+            stats.dataset_bytes_spilled = 0
+
+            def collect_stats(cur_stats):
+                stats.dataset_bytes_spilled += cur_stats.extra_metrics.get(
+                    "obj_store_mem_spilled", 0
```

Comment:
When wouldn't the `obj_store_mem_spilled` exist?

Reply:
Non-map_batches operators will not have this metric attached yet.